File: 백준/Silver/10828. 스택/스택.py
import sys
input=sys.stdin.readline
from collections import deque

n = int(input())
s = deque()
for _ in range(n):
    command = input().split()
    if command[0] == "push":
        s.append(int(command[1]))
    elif command[0] == "pop":
        if s:
            print(s.pop())
            continue
        print(-1)
    elif command[0] == "size":
        print(len(s))
    elif command[0] == "empty":
        if s:
            print(0)
            continue
        print(1)
    elif command[0] == "top":
        if s:
            print(s[-1])
            continue
        print(-1)

File: 백준/Silver/10828. 스택/스택.java
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.BufferedReader;
import java.util.StringTokenizer;
import java.util.ArrayDeque;
import java.util.Deque;

public class Main{
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        Deque<Integer> dq = new ArrayDeque<>();

        int n = Integer.parseInt(br.readLine());
        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine());
            String command = st.nextToken();

            switch (command) {
                case "push" :
                    int num = Integer.parseInt(st.nextToken());
                    dq.addLast(num);
                    break;
                case "pop" :
                    if (dq.isEmpty()) {
                        System.out.println(-1);
                    } else {
                        System.out.println(dq.removeLast());
                    }
                    break;
                case "size" :
                    System.out.println(dq.size());
                    break;
                case "empty" :
                    if (dq.isEmpty()) {
                        System.out.println(1);
                    } else {
                        System.out.println(0);
                    }
                    break;
                case "top" :
                    if (dq.isEmpty()) {
                        System.out.println(-1);
                    } else {
                        System.out.println(dq.getLast());
                    }
                    break;
            }

        }
    }
}